### PR TITLE
MIG-119 Fix pipeline 

### LINKED ISSUE #765: 'Backup' step shows up in pipeline view when running a stage migration
**Describe the bug**
I don't think this has any adverse effects, but I think we are showing a step that doesn't actually run in the pipeline view. A `stage` migration never runs the `Backup` step.

This step will never move past `Not started`.

```
    pipeline:
    - completed: "2020-11-11T17:32:16Z"
      message: Completed
      name: Prepare
      started: "2020-11-11T17:31:33Z"
    - message: Not started
      name: Backup
    - completed: "2020-11-11T17:34:09Z"
      message: Completed
      name: StageBackup
```

**To Reproduce**
1. Run a stage migration
2. Look at content of migmigration CR

**Expected behavior**
Only include pipeline steps that will run in the pipeline view.

### LINKED ISSUE #769: Progress pipeline does not show info when `CancelItinerary` is running
**Describe the bug**
I was testing #768 and noticed that while running the `CancelItinerary`, the pipeline no longer shows detailed progress info / descriptions. I think we should give more thought on how to handle `CancelItinerary` and `FailedItinerary` within the progress reporting system.

**To Reproduce**
1. Run a migration
2. Cancel the in-progress migration using the mig-ui `Cancel` button 
3. Watching the migmigration CR, notice the pipeline doesn't update.

### PR DESCRIPTION
**Pull request by pranavgaikwad**
This PR:
* Fixes issue with marking previous step complete whenever Cancel / Failed itinerary is performed 
* Renames Final Step to Cleanup 
* Removes unnecessary steps from the pipeline

Fixes #769 
Fixes #765

### CODE REVIEW

File: pkg/controller/migmigration/task.go [hunk 1]
```diff
@@ -945,13 +945,11 @@ func (t *Task) initPipeline() {
...
 
-func (t *Task) updatePipeline(prevStep string) {
-	oldStep := t.Owner.Status.FindStep(prevStep)
+func (t *Task) updatePipeline() {
 	currentStep := t.Owner.Status.FindStep(t.Step)
-	if oldStep != nil && oldStep != currentStep {
-		oldStep.MarkCompleted()
-		if t.failed() {
-			oldStep.Failed = true
+	for _, step := range t.Owner.Status.Pipeline {
```
Comment (pranavgaikwad) [inline pkg/controller/migmigration/task.go:950]: @djwhatle This makes sure all previous steps are marked complete when we move to the next step.

File: pkg/controller/migmigration/task.go [hunk 2]
```diff
@@ -927,7 +927,7 @@ func (t *Task) init() error {
 }
 
 func (t *Task) initPipeline() {
-	for _, phase := range FinalItinerary.Phases {
+	for _, phase := range t.Itinerary.Phases {
```
Comment (pranavgaikwad) [inline pkg/controller/migmigration/task.go:930]: @djwhatle This makes sure only the Steps belonging to the current itinerary are added to pipeline

File: pkg/controller/migmigration/task.go [hunk 3]
```diff
@@ -191,50 +191,50 @@ var FinalItinerary = Itinerary{
...
-		{Name: Canceling, Step: StepFinal},
-		{Name: DeleteBackups, Step: StepFinal},
-		{Name: DeleteRestores, Step: StepFinal},
-		{Name: DeleteRegistries, Step: StepFinal},
-		{Name: DeleteHookJobs, Step: StepFinal},
-		{Name: EnsureStagePodsDeleted, Step: StepFinal, all: HasStagePods},
-		{Name: EnsureAnnotationsDeleted, Step: StepFinal, any: HasPVs | HasISs},
-		{Name: Canceled, Step: StepFinal},
-		{Name: Completed, Step: StepFinal},
+		{Name: Canceling, Step: StepCleanup},
```
Comment (djwhatle) [inline pkg/controller/migmigration/task.go:203]: Name change looks good.

File: pkg/controller/migmigration/task.go (hunk 2 above)
Comment (djwhatle) [reply pkg/controller/migmigration/task.go:930]: Good catch

File: pkg/controller/migmigration/task.go (hunk 1 above)
Comment (djwhatle) [reply pkg/controller/migmigration/task.go:950]: +1

**Review by djwhatle — APPROVED** [review]
VISACK

**Pull request merged**